valhalla/jawn/src/lib/shared/request/request.ts

```diff
@@ -296,18 +295,73 @@ function truncLargeData(
   data: HeliconeRequest[],
   maxBodySize: number
 ): HeliconeRequest[] {
-  const trunced = data.map((d) => {
+  return data.map((d) => {
+    if (d.request_body && typeof d.request_body === "object") {
+      d.request_body = truncateValues(d.request_body);
+    }
+
+    if (d.response_body && typeof d.response_body === "object") {
+      d.response_body = truncateValues(d.response_body);
+    }
+
+    // Truncate the whole body if it's still too large
     return {
       ...d,
-      response_prompt: d.response_prompt,
-      request_prompt: d.request_prompt,
-      request_body: d.request_body,
-      response_body: d.response_body,
+      response_prompt:
+        JSON.stringify(d.response_prompt).length > maxBodySize
+          ? "Response prompt too large"
+          : d.response_prompt,
+      request_prompt:
+        JSON.stringify(d.request_prompt).length > maxBodySize
+          ? "Request prompt too large"
+          : d.request_prompt,
+      request_body:
+        JSON.stringify(d.request_body).length > maxBodySize
+          ? {
+              model: d.request_body?.model,
+              heliconeMessage: "Request body too large",
+              tooLarge: true,
+            }
+          : d.request_body,
+      response_body:
+        JSON.stringify(d.response_body).length > maxBodySize
+          ? {
+              heliconeMessage: "Response body too large",
+              model: d.response_body?.model,
+              tooLarge: true,
+            }
+          : d.response_body,
```

Comment:
```suggestion
        JSON.stringify(d.response_body).length > maxBodySize
          ? (!d.response_body?.streamed_data || JSON.stringify({...d.response_body, streamed_data: undefined}).length > maxBodySize ?
          {
              heliconeMessage: "Response body too large",
              model: d.response_body?.model,
              tooLarge: true,
            } : {...d.response_body, streamed_data: undefined})
            
          : d.response_body,
```

Reply:
you might want to put this in another function
but bascially, we should double check if omitting "streamed_data" causes it to be an acceptable size